benchexec/tools/ultimate.py

```diff
@@ -71,15 +73,13 @@ def project_url(self):
 
     def executable(self, tool_locator):
         exe = tool_locator.find_executable("Ultimate.py")
-        dir_name = os.path.dirname(exe)
-        logging.debug("Looking in %s for Ultimate and plugins/", dir_name)
-        for _, dir_names, file_names in os.walk(dir_name):
-            if "Ultimate" in file_names and "plugins" in dir_names:
-                return exe
-            break
+        dir_name = Path(os.path.dirname(exe))
+        logging.debug(f"Checking if {dir_name} contains a launcher jar")
+        if any([(dir_name / rel_launcher).exists() for rel_launcher in _LAUNCHER_JARS]):
```

Comment:
Does the new check also work for all the Ultimate versions for which the old check works?

Reply:
Yes, all the Ultimate versions should have a launcher JAR. Otherwise, even if the check would work, the actual invocation of Ultimate would fail.